File: BlogClient/src/app/Models/Posts/UpdatePostRequestModel.ts
import { BasePostViewModel } from "./BasePostViewModel";

export class UpdatePostRequestModel extends BasePostViewModel {
    id: number;

    constructor(
        id: number,
        title: string,
        content: string,
        excerpt: string,
        isPublished: boolean,
        categoryId: number,
        tagId: number
    ) {
        super(title, content, excerpt, isPublished, categoryId, tagId);
        this.id = id;
    }
}

File: BlogApp/BlogApp.ViewModels/RequestModels/Posts/UpdatePostRequestModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BlogApp.ViewModels.RequestModels.Posts
{
    public class UpdatePostRequestModel
    {
        public int Id { get; set; }

        public string Title { get; set; }
        public string Content { get; set; }
        public string Excerpt { get; set; }
        public bool IsPublished { get; set; }
    }
}
